src/app/shared/course.service.ts

```diff
@@ -1,19 +1,22 @@
+import { Injectable } from '@angular/core';
+
 import { courses } from './data';
 import { Course } from './course';
 
+@Injectable()
 export class CourseService {
-  courses: Course[] = courses;
+  courses: any[] = courses;
```

Comment:
Why `any`? Don't courses need to match a common interface?

Reply:
I wanted to use the `courses` array as a data source instead of a server. I think, the `Course[]` definition is a redundant dependency in this case. I will rework the service to implement this logic. It has some issues now.